### DISCUSSION TITLE
Cannot type 's'

### DISCUSSION THREAD

**Discussion by mankoff** [manna-harbour/miryoku#153] [Q&A] [answered]
Hello. I have flashed miryoku 2x to two different MCUs on two different keyboards, and both respond the same: I cannot type `s`.

These are non-standard boards. The host OS is Linux in Dvorak layout. I flash with `-e MIRYOKU_ALPHAS=QWERTY -e MIRYOKU_EXTRA=DVORAK`.

One board is a Hillside 46 with a micro-USB pro-micro. I'm using this QMK branch https://sovran.dev/mh/qmk_firmware/commit/c9120ea8e9b9014882961ba8ee149d12fe1cb982 and flashing with `qmk flash -kb 46 -km manna-harbour_miryoku -e MIRYOKU_ALPHAS=QWERTY -e MIRYOKU_EXTRA=DVORAK  -bl avrdude`.

The other board is non-standard custom with an Elite C flashed with `qmk flash -kb mankoffskey -km manna-harbour_miryoku -e MIRYOKU_ALPHAS=QWERTY -e MIRYOKU_EXTRA=DVORAK`

Both print `-` when I press my right pinky home, but should (I think?) type `s`. When I press both R and L pinky home, then `sa` is printed.

When I set by OS to US layout and flash with `qmk flash -kb 46 -km manna-harbour_miryoku -e MIRYOKU_ALPHAS=QWERTY` then I still have the same problem: Right pinky is `'` instead of `;`. Off by one.

Running `xev` I see that right pinky (`s` in Dvorak, `;` in QWERTY) is keycode 47, and the one to the right of that (`-` or `'`) is keycode 48. Should I fix this in `.Xmodmap` files? Am I doing something wrong with my flash or expectations of Miryoku?

**Comment by manna-harbour** [comment, marked as answer]
In general, the host needs to be set to US ANSI, as that is what QMK assumes. If you want to use a different layout on the host you can [customise layers](https://github.com/manna-harbour/miryoku/discussions/85)  and [substitute keycodes](https://docs.qmk.fm/#/reference_keymap_extras?id=header-files) as needed.

The QWERTY alphas option is just the alphas arrangement and is not designed for use with alternative layouts on the host. See https://github.com/manna-harbour/miryoku/issues/35. 

All alphas options have apostrophe on Base and semicolon on Num, see https://github.com/manna-harbour/miryoku/tree/master/docs/reference#base and https://github.com/manna-harbour/miryoku_babel#alphas-1.

For Dvorak, it's recommend to build with the Dvorak alphas option.


The Extra layer feature is not yet in QMK master, so for that'll you'll need to merge or build directly from the current `miryoku` branch. See https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#branches.
